java/client/src/org/openqa/selenium/io/TemporaryFilesystem.java

```diff
@@ -145,6 +145,10 @@ boolean shouldReap() {
   }
 
   public boolean deleteBaseDir() {
-    return baseDir.delete();
+    boolean wasDeleted = baseDir.delete();
+    if (wasDeleted) {
```

Comment:
@lukeis 
I also thought about it, but thought that maybe we should give the hook another chance to cleanup when the JVM exits assuming that the current attempt of deleting the `baseDir` failed. Please let me know if you still feel that this can be removed and i will update this PR with those changes as well.